log-manager/src/main/java/io/airlift/log/JsonFormatter.java

```diff
@@ -0,0 +1,47 @@
+package io.airlift.log;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.logging.Formatter;
+import java.util.logging.LogRecord;
+
+import static io.airlift.log.StaticFormatter.SYSTEM_ZONE;
+import static io.airlift.log.StaticFormatter.TIMESTAMP_FORMATTER;
+
+public class JsonFormatter
+        extends Formatter
+{
+    private static ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public String format(LogRecord record)
+    {
+        ZonedDateTime timestamp = ZonedDateTime.ofInstant(Instant.ofEpochMilli(record.getMillis()), SYSTEM_ZONE);
+
+        StringWriter stackTraceWriter = new StringWriter();
+        if (record.getThrown() != null) {
+            record.getThrown().printStackTrace(new PrintWriter(stackTraceWriter));
+        }
+
+        JsonRecord jsonRecord = new JsonRecord(
+                TIMESTAMP_FORMATTER.format(timestamp),
+                Level.fromJulLevel(record.getLevel()).name(),
+                Thread.currentThread().getName(),
+                record.getLoggerName(),
+                record.getMessage(),
+                stackTraceWriter.toString());
+
+        try {
+            return mapper.writeValueAsString(jsonRecord);
```

Comment:
Will this produce multi-line json records? We should consider producing a single entry per line for easier downstream processing, especially since the file itself is not valid json -- just a sequence of json objects.

It turns out there are a couple of popular "formats" that work in a similar way: 
* `JSONL`: https://jsonlines.org/
* ndjson: http://ndjson.org/
* JSON text sequences: https://datatracker.ietf.org/doc/html/rfc7464

https://en.wikipedia.org/wiki/JSON_streaming#Record_separator-delimited_JSON

Reply:
As long as the JSON itself is not "pretty", the values should not contain strings, since JSON strings are not permitted to have newlines or other control characters.

Reply:
It doesn't look like it should according to the jackson docs, I'll dig through the code some more to confirm. It seems like it'll use \r\n instead of a newline within values